ads/google/csa.js

```diff
@@ -0,0 +1,349 @@
+            overflow.style.display = '';
+            resizeCsa(container, currentAmpHeight - overflowHeight);
+          } else {
+            createOverflow(overflowHeight, requestedHeight, container,
+              currentAmpHeight - overflowHeight);
+          }
+        }
+      });
+
+    } else {
+
+      // If we need to backfill, make the call
+      if (backfillPageOptions != null &&
+          backfillAdblockOptions != null) {
+
+        const _googCsa = window['_googCsa'];
+
+        backfillAdblockOptions['adLoadedCallback'] =
+          generateCallback(null, null);
+
+        // Call AFS
+        _googCsa('ads', backfillPageOptions, backfillAdblockOptions);
+
+      } else {
+
+        // Let AMP know we didn't return anything
+        window.context.noContentAvailable();
+
+      }
+    }
+  };
+
+  return resizeIframe;
+}
+
+/**
+ * Helper function to resize the height of a CSA container and its child iframe
+ * @param {Object} container HTML element of the CSA container
+ * @param {number} height Height to resize, in pixels
+ */
+function resizeCsa(container, height) {
+  container.firstChild.style.height = height + 'px';
+  container.style.height = height + 'px';
+}
+
+/**
+ * Helper function to create the base overflow element
+ * @param {number} height Height of the overflow element
+ * @return {Node}
+ */
+function getOverflowElement(height) {
+  const overflow = document.createElement('div');
+  overflow.id = 'overflow';
+  overflow.style.position = 'absolute';
+  overflow.style.height = height + 'px';
+  overflow.style.width = '100%';
+  return overflow;
+}
+
+/**
+ * Helper function to create a line element for the overflow element
+ * @return {Node}
+ */
+function getOverflowLine() {
+  const line = document.createElement('div');
+  line.style.background = 'rgba(0,0,0,.16)';
+  line.style.height = '1px';
+  return line;
+}
+
+/**
+ * Helper function to create a chevron element for the overflow element
+ * @return {Node}
+ */
+function getOverflowChevron() {
+  const svg = '<svg xmlns="http://www.w3.org/2000/svg" width="36px" ' +
+      'height="36px" viewBox="0 0 48 48" fill="#757575"><path d="M14.83' +
+      ' 16.42L24 25.59l9.17-9.17L36 19.25l-12 12-12-12z"/>' +
+      '<path d="M0-.75h48v48H0z" fill="none"/> </svg>';
+
+  const chevron = document.createElement('div');
+  chevron.style.width = '36px';
+  chevron.style.height = '36px';
+  chevron.style.marginLeft = 'auto';
+  chevron.style.marginRight = 'auto';
+  chevron.style.display = 'block';
+  chevron.innerHTML = svg;
+  return chevron;
+}
+
+/**
+ * Helper function to create an overflow element
+ * @param {number} overflowH Height of the overflow element
+ * @param {number} fullH Height the iframe should be when overflow is clicked
+ * @param {Node} container HTML element of the CSA container
+ * @param {number} containerH Height to resize the CSA container, in pixels
+ */
+function createOverflow(overflowH, fullH, container, containerH) {
```

Comment:
nit: please swap the position of `createOverFlow` and `resizeCsa`. Easier for reader to follow the logic.